src/impl/apple/keychain_helper.cpp

```diff
@@ -109,21 +109,35 @@ void set_key(const std::vector<char>& key, CFStringRef account, CFStringRef serv
 
 }   // anonymous namespace
 
-std::vector<char> metadata_realm_encryption_key(bool check_legacy_service)
+std::vector<char> metadata_realm_encryption_key(bool check_legacy_service, bool recheck_legacy_service)
 {
     CFStringRef account = CFSTR("metadata");
     CFStringRef legacy_service = CFSTR("io.realm.sync.keychain");
 
+    bool skip_legacy_recheck = false;
     CFPtr<CFStringRef> service;
     if (CFStringRef bundle_id = CFBundleGetIdentifier(CFBundleGetMainBundle()))
         service = adoptCF(CFStringCreateWithFormat(NULL, NULL, CFSTR("%@ - Realm Sync Metadata Key"), bundle_id));
     else {
         service = retainCF(legacy_service);
+        skip_legacy_recheck = true;
         check_legacy_service = false;
     }
 
     // Try retrieving the key.
     if (auto existing_key = get_key(account, service.get())) {
+        if (recheck_legacy_service && !skip_legacy_recheck) {
+            // 2.10.0 introduced a bug where a new key might be generated even though
```

Comment:
I think it'd be preferable to reference a commit or PR here rather than a Cocoa version number.